rewrite-maven/src/test/java/org/openrewrite/maven/AddPluginTest.java

```diff
@@ -361,4 +362,135 @@ void addPluginWithExistingPlugin() {
           )
         );
     }
+
+    @Test
+    void addPluginOnlyToRootPom() {
+        rewriteRun(
+          spec -> spec.recipe(new AddPlugin("org.springframework.boot", "spring-boot-maven-plugin", "3.1.5", null, null, null, null, true)),
+          pomXml(
+            """
+              <project>
+                <groupId>com.mycompany.app</groupId>
+                <artifactId>my-app</artifactId>
+                <version>1</version>
```

Comment:
Can we add another test here where the parent in the repository has an external parent like Spring Boot? I'd want to make sure we still add a plugin in such common cases.

Reply:
I completely missed your comment, sorry! Added the testcase